lts_array/lts_array_plot.py

```diff
@@ -29,86 +29,101 @@ def lts_array_plot(st, lts_vel, lts_baz, t, mdccm, stdict):
             ``axs1``: Output figure axes.
 
     Example:
-        fig1, axs1 = lts_array_plot(st, stdict, t, mdccm, lts_vel, lts_baz)
+        fig, axs = lts_array_plot(st, lts_vel, lts_baz, t, mdccm, stdict)
     '''
 
     # Specify the colormap.
-    CM = 'RdYlBu_r'
+    cm = 'RdYlBu_r'
     # Colorbar/y-axis limits for MdCCM.
-    CAX = (0.2, 1)
+    cax = (0.2, 1)
     # Specify the time vector for plotting the trace.
     tvec = st[0].times('matplotlib')
 
+    # Determine the number and order of subplots.
+    num_subplots = 3
+    vplot = 1
+    bplot = 2
+    splot = bplot
+    if stdict is not None:
+        num_subplots += 1
+        splot = bplot + 1
+
     # Start plotting.
-    fig1, axarr1 = plt.subplots(4, 1, sharex='col')
-    fig1.set_size_inches(9, 12)
-    axs1 = axarr1.ravel()
-    axs1[0].plot(tvec, st[0].data, 'k')
-    axs1[0].axis('tight')
-    axs1[0].set_ylabel('Pressure [Pa]')
-    axs1[0].text(0.15, 0.93, st[0].stats.station, horizontalalignment='center',
-                 verticalalignment='center', transform=axs1[0].transAxes)
-    cbaxes = fig1.add_axes(
-        [0.9, axs1[2].get_position().y0, 0.02,
-         axs1[1].get_position().y1 - axs1[2].get_position().y0])
+    fig, axarr = plt.subplots(num_subplots, 1, sharex='col')
+    fig.set_size_inches(9, 12)
+    axs = axarr.ravel()
+    axs[0].plot(tvec, st[0].data, 'k')
+    axs[0].axis('tight')
+    axs[0].set_ylabel('Pressure [Pa]')
+    axs[0].text(0.15, 0.93, st[0].stats.station, horizontalalignment='center',
+                verticalalignment='center', transform=axs[0].transAxes)
+    cbaxes = fig.add_axes(
+        [0.95, axs[splot].get_position().y0, 0.02,
+         axs[vplot].get_position().y1 - axs[splot].get_position().y0])
 
     # Plot the trace velocity plot.
-    sc = axs1[1].scatter(t, lts_vel, c=mdccm,
-                         edgecolors='gray', lw=0.1, cmap=CM)
-    axs1[1].set_ylim(0.15, 0.60)
```

Comment:
Why 0.15-0.6? Anything other than ~0.25-0.45 is not realistic.

Reply:
> > > When ALPHA=1.0 in the example, stdict is not "None" as expected. Instead it returns "{'size': 6}".
> > 
> > 
> > This was actually done on purpose to preserve the output types, i.e. a dictionary is consistently returned. In the case of ordinary least squares, it's an empty dictionary. As in the array_processing plotting script, I expect the plotting default stdict=None behavior to be used when the station pair plot is undesired. For least squares, the output stdict variable can be ignored entirely by assigning it to `__` in the tuple unpacking.
> > Does that make sense?
> 
> OK, but then ` if stdict is not None:` on line 87 of lts_array_plot.py does not pass and it still tries to plot the element pairs subplot. Try the example with alpha=1.0 and you'll see what I mean.

Are you still passing a station dictionary as an argument to the plotting script? That will not evaluate as `None`, so the plotting script will attempt to plot the empty dictionary. I can fix this by adding an additional check for empty dictionaries. This should be done for the `array_processing` plotting script as well.

Reply:
Fixed!